src/test/fuzz/mini_miner.cpp

```diff
@@ -0,0 +1,180 @@
+            } else {
+                available_coins.push_back(COutPoint{tx->GetHash(), n});
+            }
+        }
+
+        if (fuzzed_data_provider.ConsumeBool() && !tx->vout.empty()) {
+            // Add outpoint from this tx (may or not be spent by a later tx)
+            outpoints.push_back(COutPoint{tx->GetHash(),
+                                          (uint32_t)fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, tx->vout.size())});
+        } else {
+            // Add some random outpoint (will be interpreted as confirmed or not yet submitted
+            // to mempool).
+            auto outpoint = ConsumeDeserializable<COutPoint>(fuzzed_data_provider);
+            if (outpoint) outpoints.push_back(*outpoint);
+        }
+
+    }
+
+    node::MiniMiner mini_miner{pool, outpoints};
+    assert(mini_miner.IsReadyToCalculate());
+    const CFeeRate target_feerate{CFeeRate{ConsumeMoney(fuzzed_data_provider, /*max=*/MAX_MONEY/1000)}};
+    if (fuzzed_data_provider.ConsumeBool()) {
+        const auto bump_fees = mini_miner.CalculateBumpFees(target_feerate);
+        assert(bump_fees.size() == outpoints.size());
+        for (const auto& [outpoint, fee] : bump_fees) assert(fee >= 0);
+    } else {
+        const auto bump_fee = mini_miner.CalculateTotalBumpFees(target_feerate);
+        assert(bump_fee.has_value());
+    }
+    assert(!mini_miner.IsReadyToCalculate());
+}
+
+// Test that MiniMiner and BlockAssembler build the same block given the same transactions and constraints.
+FUZZ_TARGET_INIT(mini_miner_selection, initialize_miner)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    CTxMemPool pool{CTxMemPool::Options{}};
+    // Make a copy to preserve determinism.
+    std::deque<COutPoint> available_coins = g_available_coins;
+    std::vector<CTransactionRef> transactions;
+
+    LOCK2(::cs_main, pool.cs);
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 100)
+    {
+        CMutableTransaction mtx = CMutableTransaction();
+        const size_t num_inputs = 2;
+        const size_t num_outputs = fuzzed_data_provider.ConsumeIntegralInRange<size_t>(2, 5);
+        for (size_t n{0}; n < num_inputs; ++n) {
+            auto prevout = available_coins.front();
+            mtx.vin.push_back(CTxIn(prevout, CScript()));
+            available_coins.pop_front();
+        }
+        for (uint32_t n{0}; n < num_outputs; ++n) {
+            mtx.vout.push_back(CTxOut(100, P2WSH_OP_TRUE));
+        }
+        CTransactionRef tx = MakeTransactionRef(mtx);
+
+        // First 2 outputs are available to spend. The rest are added to outpoints to calculate bumpfees.
+        // There is no overlap between spendable coins and outpoints passed to MiniMiner because the
+        // MiniMiner interprets spent coins as to-be-replaced and excludes them.
+        for (uint32_t n{0}; n < num_outputs - 1; ++n) {
+            if (fuzzed_data_provider.ConsumeBool() && fuzzed_data_provider.ConsumeBool()) {
```

Comment:
```suggestion
            if (fuzzed_data_provider.ConsumeBool()) {
```

Shouldn't really make a difference since the fuzzer picks both bools.